python/tank/path_cache.py

```diff
@@ -386,18 +386,12 @@ def synchronize(self, full_sync=False):
                 # nothing has changed since the last sync
                 log.debug("Path cache syncing not necessary - local folders already up to date!")
                 return []
-            
-            elif num_deletions > 0:
-                # some stuff was deleted. fall back on full sync
-                log.debug("Deletions detected, doing full sync")
-                return self._do_full_sync(c)
-            
-            elif num_creations > 0:
-                # we have a complete trail of increments. 
+            elif num_creations > 0 and num_deletions > 0:
```

Comment:
I've decided to keep the enumeration of deleted and created entries for debugging purpose.